stride/physics/boundaries/devito.py

```diff
@@ -5,6 +5,43 @@
 from .boundary import Boundary
 
 
+class SpongeBoundaryElastic(Boundary):
+
+    def __init__(self, grid):
+        '''
+        Sponge boundary for elastic codes
+        :param parameter: 0.2763 for homogeneous water, 0.008635 alpha2D
+        '''
+        super().__init__(grid)
+        print('======== Elastic')
+
+        self.damp = None
+
+    def apply(self, field, velocity, parameter):
+        # TODO: calculate boundary coefficient
+        space = self._grid.space
+        time = self._grid.time
+
+        self.damp = devito.Function(name="damp", grid=self._grid.devito_grid)
+        eqs = [devito.Eq(self.damp, 1.0)]
+        padsizes = [space.absorbing for _ in range(self._grid.devito_grid.dim)]
+        for (nbl, nbr), d in zip(padsizes, self.damp.dimensions):
+            dampcoeff = parameter * space.spacing[0]  # 0.2763102111592855 (water_example)
+            # left
+            dim_l = devito.SubDimension.left(name='abc_%s_l' % d.name, parent=d, thickness=nbl)
+            pos = devito.Abs((nbl - (dim_l - d.symbolic_min) + 1) / float(nbl))
+            val = - dampcoeff * (pos - devito.sin(2 * np.pi * pos) / (2 * np.pi))
+            eqs += [devito.Inc(self.damp.subs({d: dim_l}), val / d.spacing)]
+            # right
+            dampcoeff = parameter * space.spacing[0]  # 0.2763102111592855 (water_example)
+            dim_r = devito.SubDimension.right(name='abc_%s_r' % d.name, parent=d, thickness=nbr)
+            pos = devito.Abs((nbr - (d.symbolic_max - dim_r) + 1) / float(nbr))
+            val = - dampcoeff * (pos - devito.sin(2 * np.pi * pos) / (2 * np.pi))
+            eqs += [devito.Inc(self.damp.subs({d: dim_r}), val / d.spacing)]
+        devito.Operator(eqs, name='initdamp')()
```

Comment:
You can (and should) populate the "self.damp" function using the method "self.damping". You can find it here: https://github.com/trustimaging/stride/blob/de3bcf1bea6ccd7125a753c1728b51fcadd43db0/stride/physics/boundaries/boundary.py#L59

In this case, you should select " damping_type='sine' ".

Getting the damping layer that you expect might take some experimentation, so let me know if you have any problems with that.

Reply:
You should also check in case there is no boundary layer at all:

```
if np.max(space.extra) > 0:
    # calculate self.damp
else:
    self.damp = 1
```

Reply:
Replace code in SpongeBoundary1 with lines 57-66 (i.e. SpongeBoundary2). Remove scaling factor of 7, and include reflection_coefficient code from lines 88-89.

Reply:
Implemented